core/src/services/dropbox/core.rs

```diff
@@ -0,0 +1,149 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use std::collections::HashMap;
+use std::default::Default;
+use std::fmt::Debug;
+use std::fmt::Formatter;
+use std::rc::Rc;
+
+use crate::raw::percent_encode_path;
+use crate::raw::HttpClient;
+use crate::raw::{new_json_deserialize_error, RpCopy};
+use crate::Error;
+use crate::ErrorKind;
+
+use bytes::Bytes;
+use http::request::Builder;
+use http::StatusCode;
+use http::{header, Request, Response};
+use serde::{Deserialize, Serialize};
+use tokio::sync::Mutex;
+
+use crate::{
+    raw::{new_request_build_error, AsyncBody, IncomingAsyncBody},
+    types::Result,
+};
+
+pub struct DropboxCore {
+    pub token: String,
+    pub client: HttpClient,
+    pub root: String,
+}
+
+impl Debug for DropboxCore {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        let mut de = f.debug_struct("DropboxCore");
+        de.finish()
+    }
+}
+
+impl DropboxCore {
+    pub async fn dropbox_get(&self, path: &str) -> Result<Response<IncomingAsyncBody>> {
+        let url: String = "https://content.dropboxapi.com/2/files/download".to_string();
+        let download_args = DropboxDownloadArgs {
+            path: format!("{}{}", self.root, path).to_string(),
+        };
+        let request = self
+            .sign(Request::post(&url))
+            .header(
+                "Dropbox-API-Arg",
+                serde_json::to_string(&download_args).unwrap(),
+            )
+            .body(AsyncBody::Empty)
+            .map_err(new_request_build_error)?;
+        self.client.send(request).await
+    }
+
+    pub async fn dropbox_update(
+        &self,
+        path: &str,
+        size: Option<usize>,
+        content_type: Option<&str>,
+        body: AsyncBody,
+    ) -> Result<Response<IncomingAsyncBody>> {
+        let url = "https://content.dropboxapi.com/2/files/upload".to_string();
+        let mut args = DropboxUploadArgs::default();
+        args.path = format!("{}{}", self.root, path).to_string();
```

Comment:
Please build args in direct way like:

```rust
let args = DropboxUploadArgs { xx: yy }
```

Structs like `DropboxUploadArgs` are internally used only in dropbox mod, so it's ok to make all its field public.

Reply:
The UploadArgs has to many default value, so I prefer to init it as `::default()`

Reply:
Please use the following style instead.

```rust
Args {
  xx: yy
  ..Default::default()
}
```